spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java

```diff
@@ -0,0 +1,289 @@
+import org.springframework.integration.support.AbstractIntegrationMessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.HeaderMapper;
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory;
+import org.springframework.util.Assert;
+
+/**
+ * Debezium Change Event Channel Adapter.
+ *
+ * @author Christian Tzolov
+ * @since 6.2
+ */
+public class DebeziumMessageProducer extends MessageProducerSupport {
+
+	private DebeziumEngine.Builder<ChangeEvent<byte[], byte[]>> debeziumEngineBuilder;
+
+	private DebeziumEngine<ChangeEvent<byte[], byte[]>> debeziumEngine;
+
+	/**
+	 * Executor service for running engine daemon.
+	 */
+	private ExecutorService executorService = Executors
+			.newSingleThreadExecutor(new CustomizableThreadFactory("debezium-"));
+
+	/**
+	 * Flag to denote whether the {@link ExecutorService} was provided via the setter and thus should not be shutdown
+	 * when {@link #destroy()} is called.
+	 */
+	private boolean executorServiceExplicitlySet = false;
+
+	private CountDownLatch latch = new CountDownLatch(0);
+
+	private Future<?> future = CompletableFuture.completedFuture(null);
+
+	private String contentType = "application/json";
+
+	private HeaderMapper<List<Header<Object>>> headerMapper = new DefaultDebeziumHeaderMapper();
+
+	private boolean enableEmptyPayload = false;
+
+	private boolean enableBatch = false;
+
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
+		Assert.notNull(debeziumBuilder, "Failed to resolve Debezium Engine Builder. " +
+				"Debezium Engine Builder must either be set explicitly via constructor argument.");
+		this.debeziumEngineBuilder = debeziumBuilder;
+	}
+
+	/**
+	 * Comma-separated list of names of Debezium's Change Event headers to be mapped to the outbound Message headers.
+	 *
+	 * The Debezium' New Record State Extraction 'add.headers' property is used to configure the metadata to be set in
+	 * the produced ChangeEvent headers.
+	 *
+	 * Note that you must prefix the 'headerNames' used the 'setAllowedHeaderNames' with the prefix configured by the
+	 * 'add.headers.prefix' debezium property. Later defaults to '__'. For example for 'add.headers=op,name' and
+	 * 'add.headers.prefix=__' you should use headerNames == "__op", "__name".
+	 *
+	 * @param headerNames The values in this list can be a simple patterns to be matched against the header names (e.g.
+	 * "foo*" or "*foo").
+	 *
+	 * @see <a href=
+	 * "https://debezium.io/documentation/reference/2.2/transformations/event-flattening.html#extract-new-record-state-add-headers">add.headers</a>
+	 * @see <a href=
+	 * "https://debezium.io/documentation/reference/2.2/transformations/event-flattening.html#extract-new-record-state-add-headers-prefix">add.headers.prefix</a>
+	 *
+	 */
+	public void setAllowedHeaderNames(String... headerNames) {
```

Comment:
I don't think this is a channel adapter responsibility.
Please, consider to move this into `DefaultDebeziumHeaderMapper` impl.
Well, I see you have it there already anyway.
But then just a setter for a `HeaderMapper` must be provided here on adapter.